### ISSUE TITLE
Unable to return to original desktop with Windows icons and menus after Libki sign in

### ISSUE THREAD

**Issue by Eng-Luke** [Libki/libki-client#98]
**Describe the bug**
I am using Libki on my Windows computer and after logging out of a remote session, I am unable to return to my original desktop with Windows icons and menus. Instead, I am left with a blank screen. I have tried restarting my computer and reinstalling Libki, but the issue persists. I would appreciate any help in resolving this issue.

Operating system: Windows 10

Libki version: 4.0.2

**Comment by kylemhall** [comment]
Can you provide some more detail? Is your client set to reboot, log out, or do nothing? For production it should be set to reboot or log out.

**Comment by Eng-Luke** [comment]
Yes so i set it to do nothing the last option.  Its set to do nothing. Okay so will it also return the desktop screen then with Windows Icons and Menus

**Comment by kylemhall** [comment]
I'm not sure what you mean by "remote session". Could you explain?

What happens is Libki stops explorer.exe when it launches, then on login it restarts explorer.exe.
This should restore your desktop and start bar. It sounds like on your system it isn't. The behavior you describe is consistent with explorer.exe already running. If explorerer.exe is run while it is already running, it just opens an explorer window on your desktop.

What is confounding is that you still have no start bar or desktop items. I don't think this issue is directly related to Libki. I've not seen this behavior before.

If, when you have no destkop or taskbar, you run explorer.exe from the task menu, does your desktop come back?

**Comment by Eng-Luke** [comment]
So when I have no destkop or taskbar, and I run explorer.exe from the task menu, my desktop does not come back. It will simply open explorer.exe which launches a File Explorer that starts. Find attached Screenshots of the current layout. So if a student closes the file explorer they face challenges in trying to access Apps like MS Word and Chrome for the interface will just be black. So all i want is for the students to see Windows, Icons and Menus everytime they logon. I am using laptop computers about 50 at most 
![After running explorer](https://user-images.githubusercontent.com/129087829/228146628-2dac3782-56fb-42e7-bc08-04cc1c90d5a2.PNG)
![Blank Interface](https://user-images.githubusercontent.com/129087829/228146646-91d188fd-9196-4611-a1dc-579b9396e688.PNG)
![Interface 1](https://user-images.githubusercontent.com/129087829/228146651-fc4ddf16-764d-4635-abac-bcd5ff21bec2.PNG)
![Blank Interface](https://user-images.githubusercontent.com/129087829/228146732-e9da2b13-350b-4f5e-8eb7-2ba062be6c69.PNG)
![Interface 1](https://user-images.githubusercontent.com/129087829/228146746-f8dad732-7488-464a-be7c-471415d8f993.PNG)
![After running explorer](https://user-images.githubusercontent.com/129087829/228146755-0d1e2b71-b17f-4a11-a752-3a7fc53e9bf3.PNG)

**Comment by kylemhall** [comment]
Have you configured Libki to run as the user's login shell, or just as a startup item?

**Comment by Eng-Luke** [comment]
I did not get that option but the first 30 we set up it was set to " Automatically start libki client after normal user shell" Then for others we used "Automatically start Libki instead of user shell(shell replacement)"

**Comment by kylemhall** [comment]
When you installed the client, was it from an admin account, or from the same account that the client is meant to be used from? I think there may be a bug in the installer. If you've been using an admin account, can you try installing with shell replacement from the account that is meant to use the client?

**Comment by cbajak** [comment]
It was originally installed from the Admin account. I uninstalled and tried again from a limited account and same thing. Computer boots to windows login screen, I log in as Public in this case (no password), screen is black, a quick CMD box pops up … killtask.exe, then the libki screen. I enter my password and am taken to a black screen with file explorer window open. 

I will pull another machine off the shelf in the next couple of days, try again on a clean W10 install, then get back to you.

**Comment by loidor** [comment]
It sounds like something is wrong with the shell registry. Don't remember the exact path right now, but I think it's current_machine/software/microsoft/windows nt/winlogon. Check what's in the shell value and remove everything but explorer.exe.

31 Mar 2023 18:10:17 cbajak ***@***.***>:

**Comment by cbajak** [comment]
Ok... I replaced everything on that line with just explorer.exe. I have my desktop back. :)
